src/lib/util/lists.ml

```diff
@@ -62,3 +66,13 @@ let to_seq l =
   in
   aux l
 
+let rec compare cmp l1 l2 =
```

Comment:
This is defined in OCaml's standard library as `List.compare`, but sadly only from `4.12`. Maybe leave a `TODO` to remove this and use the standard `List.compare` in the future?

Reply:
I know but sure I will add a comment :)